File: решения/java/src/Lab15/task6876/Comparator6876.java
package Lab15.task6876;

import java.util.Comparator;

/**
 * @author akryukov
 *         19.07.2017
 */
public class Comparator6876 implements Comparator<Shape> {
    @Override
    public int compare(Shape o1, Shape o2) {
        double area1 = o1.area();
        double area2 = o2.area();

        return (int) (area2 - area1);
    }
}

File: решения/cs/Classes/task6876/Comparator6876.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Classes.Task6876
{
    public class Comparator6876 : Comparer<Shape>
    {
        public override int Compare(Shape o1, Shape o2)
        {
            return (int)(o2.Area - o1.Area);
        }
    }
}
